modules/fluid_properties/test/tests/interfaces/nan_interface/tests

```diff
@@ -6,33 +6,33 @@
     method = 'DBG'
     threading = '!pthreads'
 
-    requirement = 'MOOSE shall provide an interface for using either quiet or signaling NaNs'
+    requirement = 'MOOSE shall provide an interface for producing errors, warnings, or just quiet NaNs'
     design = '/NaNInterface.md'
-    issues = '#12234'
+    issues = '#12234 #12350'
   [../]
   [./signaling_nan_dbg]
     type = 'RunApp'
     input = 'nan_interface.i'
-    cli_args = 'Modules/FluidProperties/fp/use_quiet_nans=false'
+    cli_args = 'Modules/FluidProperties/fp/emit_on_nan=error'
     should_crash = True
     allow_test_objects = True
     method = 'DBG'
     threading = '!pthreads'
 
-    requirement = 'MOOSE shall provide an interface for using either quiet or signaling NaNs'
+    requirement = 'MOOSE shall provide an interface for producing errors, warnings, or just quiet NaNs'
     design = '/NaNInterface.md'
-    issues = '#12234'
+    issues = '#12234 #12350'
   [../]
   [./signaling_nan_opt]
     type = 'RunApp'
     input = 'nan_interface.i'
-    cli_args = 'Modules/FluidProperties/fp/use_quiet_nans=false'
+    cli_args = 'Modules/FluidProperties/fp/emit_on_nan=error'
```

Comment:
This is not correct. Originally, we would produce a NaN, but in opt trapping FPE is off, so the test finished as expected.  This change makes this test to fail, because it will produce a `mooseError` which is alway a hard fail.

What I think you should be doing here is to _not_ pass any arguments, so that `emit_on_nan` will use the default value.  Similarly, you should not be  setting this parameter in the `signaling_nan_dbg` test. That is the point of these 2 tests: when people run in opt, we do not error out, but we do in debug.

With this in mind I think the requirements should be changed like so:

* `signaling_nan_dbg`: "The system should report an error when a NaN is produced by a computation in DEBUG mode, by default".
* `signaling_nan_opt`: "The system should not report an error when a NaN is produced by a computation in OPT mode, by default".

And I am also thinking we should change `quiet_nan` to `warning_on_nan` where we pass `emit_on_nan=warning`, because we are not testing that at all. The test would not have the method restriction and its `requirement` would be "The system should produce a warning when a NaN is produced and user required that the execution would not terminate."